src/foam/nanos/notification/EmailSetting.js

```diff
@@ -63,19 +65,20 @@ foam.CLASS({
           if ( ! disabledTopics.contains(notification.getNotificationType()) ) {
             EmailMessage message = new EmailMessage();
             message.setTo(new String[]{user.getEmail()});
+            Notification clone = (Notification) notification.fclone();
 
-            if ( notification.getEmailArgs() != null ) {
-              Map<String, Object> emailArgs = resolveNotificationArguments(x, notification.getEmailArgs(), user);
-              notification.setEmailArgs(emailArgs);
+            if ( clone.getEmailArgs() != null ) {
+              Map<String, Object> emailArgs = resolveNotificationArguments(x, clone.getEmailArgs(), user);
+              clone.setEmailArgs(emailArgs);
             }
 
             try {
-              if ( foam.util.SafetyUtil.isEmpty(notification.getEmailName()) ) {
-                message.setSubject(notification.getTemplate());
-                message.setBody(notification.getBody());
+              if ( foam.util.SafetyUtil.isEmpty(clone.getEmailName()) ) {
```

Comment:
sorry didn't catch this on previous reviews but,
this whole segment seems unnecessary:
 'if ( foam.util.SafetyUtil.isEmpty(clone.getEmailName()) '
This check is done in the EmailsUtility.
so passing clone.getEmailName() is fine no matter what.

The user, being passed in is for group usage. Don't see why you would want to pass in 'null'
Further if there is no template, then the args being passed in are ignored - or at least should be.

Reply:
To give some context on Notification Settings as an implementation. Users will be able to receive notifications based on what settings they own. As a developer I might create a notification and pass it into a users doNotify method. When that method is called it will attempt to send notifications based on all notification settings the user may have/opted into. For example (Email, SMS or Push).

Looks like the condition in EmailsUtility prevents this logic from happening if no email template is passed in. Which we'll have to allow.